### PR TITLE
Simple server side refactor.

### PR DESCRIPTION
**Pull request by KPP13**
Duplicates removed. Logs added to report exceptions and errors.

### CODE REVIEW

File: server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java [hunk 1]
```diff
@@ -3,14 +3,16 @@
 import java.io.IOException;
 import java.nio.channels.SocketChannel;
 
-// TODO: change ending game - send a message from the server, when one disconnected
-// thread to manage the game of two players
```
Comment (maciejkocur) [inline server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java]: :+1:

File: server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java [hunk 2]
```diff
@@ -29,25 +30,23 @@ public void run() {
         p1.start();
         p2.start();
 
-        // important?
```
Comment (maciejkocur) [inline server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java]: :+1:

File: server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java [hunk 3]
```diff
@@ -3,14 +3,16 @@
...
-class ConnectionThread implements Runnable {
+import static org.apache.log4j.Logger.*;
+import org.apache.log4j.Logger;
 
+class ConnectionThread implements Runnable {
     private final SocketChannel sc1, sc2;
     private final int gameIndex;
     private final Game game = new Game();
 
+    private final static Logger log = getLogger(Server.class);
```
Comment (clinton56) [inline server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java]: Copy paste error :dancers:

File: server/src/main/java/io/github/expansionteam/battleships/PlayerThread.java [hunk 4]
```diff
@@ -50,26 +48,13 @@ public void run() {
...
-//        }
         catch (IOException e) {
-            e.printStackTrace();
+            log.error("FAILED", e);
         } finally {
             try {
                 socketChannel.close();
             } catch (IOException e) {
-                e.printStackTrace();
+                log.error("FAILED", e);
```
Comment (clinton56) [inline server/src/main/java/io/github/expansionteam/battleships/PlayerThread.java]: Why do you always use "FAILED" as error message?

**Comment by maciejkocur** [comment]
For me it's ready for merge.

File: server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java [hunk 5]
```diff
@@ -3,14 +3,16 @@
 import java.io.IOException;
 import java.nio.channels.SocketChannel;
 
-// TODO: change ending game - send a message from the server, when one disconnected
-// thread to manage the game of two players
-class ConnectionThread implements Runnable {
+import static org.apache.log4j.Logger.*;
+import org.apache.log4j.Logger;
 
+class ConnectionThread implements Runnable {
```
Comment (LIttleAncientForestKami) [inline server/src/main/java/io/github/expansionteam/battleships/ConnectionThread.java]: XThread usually `extends Thread`, not `implements Runnable`. Crucial difference between the two approaches to thread creation is that first (extending) makes a child-class a thread as well. The other allows for extending something else, thus leaves the type not so heavily changed.

In short: think of a better name. Perhaps `ConnectionTask`?

File: server/src/main/java/io/github/expansionteam/battleships/Game.java [hunk 6]
```diff
@@ -26,45 +26,35 @@ private boolean isFieldHit(Board board, int x, int y) {
...
-            synchronized (randomShipGenerator) {
-                randomShipGenerator.generateRandomShips(firstPlayerBoard);
-            }
-        } else {
-            synchronized (randomShipGenerator) {
-                randomShipGenerator.generateRandomShips(secondPlayerBoard);
-            }
+        Board board = currentBoard();
+        synchronized (randomShipGenerator) {
+            randomShipGenerator.generateRandomShips(board);
```
Comment (LIttleAncientForestKami) [inline server/src/main/java/io/github/expansionteam/battleships/Game.java]: Nice parametrization!

File: server/src/main/java/io/github/expansionteam/battleships/Game.java [hunk 7]
```diff
@@ -26,45 +26,35 @@ private boolean isFieldHit(Board board, int x, int y) {
...
+        return isFieldHit(board, x, y);
     }
 
     public boolean isDestroyedShip(int x, int y) {
-        if (firstPlayer()) {
-            return firstPlayerBoard.isDestroyedShip(x, y);
-        }
-        return secondPlayerBoard.isDestroyedShip(x, y);
+        Board board = currentBoard();
+        return board.isDestroyedShip(x, y);
```
Comment (LIttleAncientForestKami) [inline server/src/main/java/io/github/expansionteam/battleships/Game.java]: Is ship destroyed sounds better to me, must say.

File: server/src/main/java/io/github/expansionteam/battleships/JsonHandler.java [hunk 8]
```diff
@@ -8,12 +8,12 @@
...
 
-public class JsonHandler {
+class JsonHandler {
 
     String resolveAction(String json, Game game) {
-        JSONObject jsonRequest = new JSONObject(json);
-        String type = jsonRequest.getString("type");
-        JSONObject jsonResponse = new JSONObject();
+        final JSONObject jsonRequest = new JSONObject(json);
+        final String type = jsonRequest.getString("type");
```
Comment (LIttleAncientForestKami) [inline server/src/main/java/io/github/expansionteam/battleships/JsonHandler.java]: You can create enum out of String, if String is to be final. That is, of course, if you have anything enumerating types of events. :-)

File: server/src/main/java/io/github/expansionteam/battleships/PlayerThread.java (hunk 4 above)
Comment (LIttleAncientForestKami) [inline server/src/main/java/io/github/expansionteam/battleships/PlayerThread.java]: I'm guessing for consistency. Counter-proposal? It's good to log exceptions in uniform manner. Makes it easier to later spot them in logs.

File: server/src/main/java/io/github/expansionteam/battleships/Server.java [hunk 9]
```diff
@@ -5,47 +5,47 @@
 import java.nio.channels.ServerSocketChannel;
 import java.nio.channels.SocketChannel;
 
+import static org.apache.log4j.Logger.*;
+import org.apache.log4j.Logger;
+
 // I want to replace some code with SELECTORS, but I made stupid mistake yesterday
```
Comment (LIttleAncientForestKami) [inline server/src/main/java/io/github/expansionteam/battleships/Server.java]: Good info on class, says something about it. Convert to either `TODO` or Javadoc.

File: server/src/main/java/io/github/expansionteam/battleships/Server.java [hunk 10]
```diff
@@ -5,47 +5,47 @@
...
             }
             catch (IOException e) {
-                System.out.println("IOException...");
-                e.printStackTrace();
+                log.error("FAILED", e);
             }
         }
     }
 
     // accept connections (new clients)
```
Comment (LIttleAncientForestKami) [inline server/src/main/java/io/github/expansionteam/battleships/Server.java]: Javadoc

**Comment by LIttleAncientForestKami** [comment]
Really nice refactor, must say. Aside from few details, spotted nothing wrong. Most major remark is about renaming `ConnectionThread` as it doesn't extend Thread. I'm OK with this happening later. Merging.

**Pull request merged**